README.md
<img align="center" src="https://raw.githubusercontent.com/arcticicestudio/northem-dark-java/develop/src/main/assets/media/northem-dark-java-banner.png" alt="Northem Dark Java"/> <a href="https://github.com/arcticicestudio/northem-dark"><img align="right" src="https://raw.githubusercontent.com/arcticicestudio/northem-dark-java/develop/src/main/assets/media/northem-logo.png"/></a>

<img src="https://cdn.travis-ci.org/images/favicon-c566132d45ab1a9bcae64d8d90e4378a.svg" width=24 height=24/> [![Travis CI](https://img.shields.io/travis/arcticicestudio/northem-dark-java/develop.svg)](https://travis-ci.org/arcticicestudio/northem-dark-java) <img src="https://codecov.io/favicon.ico" width=24 height=24/> [![codecov](https://codecov.io/gh/arcticicestudio/northem-dark-java/branch/develop/graph/badge.svg)](https://codecov.io/gh/arcticicestudio/northem-dark-java)  
<img src="https://assets-cdn.github.com/favicon.ico" width=24 height=24/> [![release](https://img.shields.io/github/release/arcticicestudio/northem-dark-java.svg)](https://github.com/arcticicestudio/northem-dark-java/releases/latest) <img src="http://search.maven.org/favicon.ico" width=24 height=24/> [![Central Repository](https://img.shields.io/maven-central/v/com.arcticicestudio/northem-dark-java.svg)](http://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22com.arcticicestudio%22%20AND%20a%3A%22northem-dark-java%22) [![OSS Sonatype Snapshot](https://img.shields.io/badge/snapshot-0.3.0--SNAPSHOT-blue.svg)](https://oss.sonatype.org/content/repositories/snapshots/com/arcticicestudio/northem-dark-java)

Java library implementation of the north-bluish dark color palette [Northem Dark](https://github.com/arcticicestudio/northem-dark).  
Optimized for syntax highlighting- and UI projects.  
This is a darker and more blue-tinged variation.

## Getting started
### Setup
To use northem-dark-java it must be available on your classpath.  
You can get it from the <img src="https://search.maven.org/favicon.ico" width=16 height=16/> <a href="https://search.maven.org">Central Repository</a> as a dependency for your favorite build tool or [download the latest version](https://github.com/arcticicestudio/northem-dark-java/releases/latest).

<img src="http://apache.org/favicons/favicon.ico" width=16 height=16/> <a href="https://maven.apache.org">Apache Maven</a>
```xml
<dependency>
  <groupId>com.arcticicestudio</groupId>
  <artifactId>northem-dark-java</artifactId>
  <version>0.2.0</version>
</dependency>
```

<img src="https://gradle.org/wp-content/uploads/fbrfg/favicon.ico" width=16 height=16/> <a href="https://gradle.org">Gradle</a>
```java
compile(group: 'com.arcticicestudio', name: 'northem-dark-java', version: '0.2.0')
```

<img src="http://apache.org/favicons/favicon.ico" width=16 height=16/> <a href="https://ant.apache.org/ivy">Apache Ivy</a>
```xml
<dependency org="com.arcticicestudio" name="northem-dark-java" rev="0.2.0" />
```

Development snapshots are available via [OSS Sonatype](https://oss.sonatype.org/content/repositories/snapshots/com/arcticicestudio/northem-dark-java).   

### Build
Build and install northem-dark-java into your local repository without GPG signing:  
```
mvn clean install
```

Signed artifacts may be build by using the `sign-gpg` profile with a provided `gpg.keyname` property:
```
mvn clean install -Dgpg.keyname=YourGPGKeyId
```

A continuous integration build is running at [Travis-CI](https://travis-ci.org/arcticicestudio/northem-dark-java).

## Variations
  - <a href="https://github.com/arcticicestudio/northem-dark-java"><img src="https://java.com/favicon.ico" alt="Java logo" width=16 height=16> Northem - Java</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" alt="Linux Logo" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" alt="Apple Logo" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" alt="Windows Logo" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-light-java"><img src="https://java.com/favicon.ico" alt="Java logo" width=16 height=16> Northem Light - Java</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" alt="Linux Logo" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" alt="Apple Logo" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" alt="Windows Logo" width=16 height=16 />

## Other Projects
  - <a href="https://github.com/arcticicestudio/northem-dark-intellij-idea-syntax"><img src="https://www.jetbrains.com/_assets//shared/favicons/jetbrains.ico" alt="JetBrain logo" width=16 height=16> Northem Dark IntelliJ IDEA Syntax</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" alt="Linux Logo" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" alt="Apple Logo" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" alt="Windows Logo" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-dark-atom-syntax"><img src="https://atom.io/favicon.ico" alt="Atom logo" width=16 height=16> Northem Dark Atom Syntax</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" alt="Linux Logo" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" alt="Apple Logo" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" alt="Windows Logo" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-dark-atom-ui"><img src="https://atom.io/favicon.ico" alt="Atom logo" width=16 height=16> Northem Dark Atom UI</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" alt="Linux Logo" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" alt="Apple Logo" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" alt="Windows Logo" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-dark-eclipse-syntax"><img src="https://eclipse.org/favicon.ico" alt="Eclipse Logo" width=16 height=16> Northem Dark Eclipse Syntax</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" alt="Linux Logo" width=16 height=16 /> <img src="https://developer.apple.com/favicon.ico" alt="Apple Logo" width=16 height=16 /> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" alt="Windows Logo" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-dark-gedit"><img src="https://static.gnome.org/wiki.gnome.org/gnome/css/favicon.png" alt="gedit Logo" width=16 height=16> Northem Dark gedit</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" alt="Linux Logo" width=16 height=16 />
  - <a href="#"><img src="http://www.gtk.org/images/gtk-logo.ico" alt="GTK Logo" width=16 height=16> Northem Dark GTK (in drafting)</a> <img src="https://www.kernel.org/theme/images/logos/favicon.png" alt="Linux Logo" width=16 height=16 />
  - <a href="https://github.com/arcticicestudio/northem-dark-notepadplusplus"><img src="http://notepad-plus-plus.org/assets/images/favicon.ico" alt="Notepadd++ Logo" width=16 height=16> Northem Dark Notepad++</a> <img src="https://www.microsoft.com/en-us/windows/favicon.ico" alt="Windows Logo" width=16 height=16 />

Visit the [official website](http://arcticicestudio.com/northem) for more information.

## Development
[![](https://img.shields.io/badge/Changelog-v0.2.0-blue.svg)](https://github.com/arcticicestudio/northem-dark-java/blob/v0.2.0/CHANGELOG.md)

### Workflow
This project follows the [gitflow](http://nvie.com/posts/a-successful-git-branching-model) branching model.

### Specifications
This project follows the [Arctic Versioning Specification (ArcVer)](https://github.com/arcticicestudio/arcver).

### Contribution
Please report issues/bugs, suggestions for improvements and feature requests to the [issuetracker](https://github.com/arcticicestudio/northem-dark-java/issues).

---

<img src="http://arcticicestudio.com/favicon.ico" width=16 height=16/> Copyright &copy; 2016 Arctic Ice Studio

[![GPL-3.0](http://www.gnu.org/graphics/gplv3-88x31.png)](http://www.gnu.org/licenses/gpl.txt) [![CC BY-SA-4.0](http://mirrors.creativecommons.org/presskit/buttons/88x31/svg/by-sa.svg)](http://creativecommons.org/licenses/by-sa/4.0/)


CHANGELOG.md
<img align="center" src="https://raw.githubusercontent.com/arcticicestudio/northem-dark-java/develop/src/main/assets/media/northem-dark-java-banner.png" alt="Northem Dark Java"/> <a href="https://github.com/arcticicestudio/northem-dark"><img align="right" src="https://raw.githubusercontent.com/arcticicestudio/northem-dark-java/develop/src/main/assets/media/northem-logo.png"/></a>

## 0.2.0 (2016-08-03) - Apache Maven Migration
This version is mainly focused on a [Apache Maven](https://maven.apache.org) migration.
Release and snapshot versions are now deployed to the [Central Repository](https://search.maven.org) via [OSS Sonatype](https://oss.sonatype.org).

### Improvements
#### Build Tool
  - Migrated to [Apache Maven](https://maven.apache.org) based on the [`glacier-apache-maven@0.3.0`](https://github.com/arcticicestudio/glacier-apache-maven) project skeletons

#### Documentation
  - The `README` has been modernized and now includes a "Getting started" section

## 0.1.0 (2016-04-09)
### Features
  - Implemented the [Northem Dark](https://github.com/arcticicestudio/northem-dark) color palette enumeration class [`NorthemDark`](https://github.com/arcticicestudio/northem-dark-java/blob/master/src/main/java/com/arcticicestudio/northem/NorthemDark.java)  

**Constants**

| Name           | RGB Value     |
| -------------- | ------------- |
| `DARKEST_COAL` | `31,31,35`    |
| `DARK_COAL`    | `36,36,41`    |
| `COAL`         | `41,41,46`    |
| `LIGHT_COAL`   | `79,79,89`    |
| `DIRTY_WHITE`  | `222,222,222` |
| `WHITE`        | `236,238,239` |
| `AQUA`         | `143,206,200` |
| `LIGHT_BLUE`   | `133,193,211` |
| `BLUE`         | `135,178,206` |
| `DARK_BLUE`    | `91,129,175`  |
| `RED`          | `224,116,115` |
| `YELLOW`       | `253,206,132` |
| `ORANGE`       | `240,160,111` |
| `LIME`         | `201,204,120` |
| `PURPLE`       | `194,161,202` |

**Methods**

| Name | Description |
| ---- | ----------- |
| `NorthemDark(final int RED, final int GREEN, final int BLUE)` | Constructs a new color object. |
| `public static String rgb(final Color COLOR)` | Converts the color to the RGB identifier. |
| `public static String hex(final Color COLOR)` | Converts the color to the HEX identifier. |
| `public Color get()` | Returns the color object. |

## 0.0.0 (2016-04-09) - Repository Reinitialization


src/test/java/com/arcticicestudio/northem/dark/NorthemDarkTest.java
/*
+++++++++++++++++++++++++++++++++++++++++++++++
title     Northem Dark Enumeration JUnit Test +
project   northem-dark-java                   +
file      NorthemDarkTest.java                +
version   0.2.0                               +
author    Arctic Ice Studio                   +
email     development@arcticicestudio.com     +
website   http://arcticicestudio.com          +
copyright Copyright (C) 2016                  +
created   2016-01-23 15:47 UTC+0100           +
modified  2016-04-09 14:08 UTC+0200           +
+++++++++++++++++++++++++++++++++++++++++++++++

[Description]
JUnit test class for the 'NorthemDark' class of the "Northem Dark - Java" API.

[Copyright]
Copyright (C) 2016 Arctic Ice Studio <development@arcticicestudio.com>

[References]
Northem Dark
  (https://github.com/arcticicestudio/northem-dark)
Java 8 API Documentation
  (https://docs.oracle.com/javase/8/docs/api/)
JavaFX 2.2 API Documentation
  (https://docs.oracle.com/javafx/2/api/)
JavaFX 2.2 Oracle Project Website
  (http://docs.oracle.com/javase/8/javase-clienttechnologies.htm)
Arctic Versioning Specification
  (http://specs.arcticicestudio.com/arcver) (ArcVer)
*/

package com.arcticicestudio.northem.dark;

import org.junit.Test;
import static org.junit.Assert.assertEquals;

/**
 * JUnit test class for the {@link NorthemDark} class of the
 * <a href="https://github.com/arcticicestudio/northem-dark-java">Northem Dark - Java</a> API.
 *
 * @author Arctic Ice Studio &lt;development@arcticicestudio.com&gt;
 * @see NorthemDark
 * @see <a href="http://junit.org/">http://junit.org/</a>
 * @see <a href="https://github.com/arcticicestudio/northem-dark-java">Northem Dark - Java</a>
 * @since 0.1.0
 */
public class NorthemDarkTest {
  @Test
  public void shouldReturnTheCorrectHEXColorCode() {
    assertEquals(NorthemDark.hex(NorthemDark.DARKEST_COAL.get()), "#1F1F23");
    assertEquals(NorthemDark.hex(NorthemDark.DARK_COAL.get()), "#242429");
    assertEquals(NorthemDark.hex(NorthemDark.COAL.get()), "#29292E");
    assertEquals(NorthemDark.hex(NorthemDark.LIGHT_COAL.get()), "#4F4F59");
    assertEquals(NorthemDark.hex(NorthemDark.DIRTY_WHITE.get()), "#DEDEDE");
    assertEquals(NorthemDark.hex(NorthemDark.WHITE.get()), "#ECEEEF");
    assertEquals(NorthemDark.hex(NorthemDark.AQUA.get()), "#8FCEC8");
    assertEquals(NorthemDark.hex(NorthemDark.LIGHT_BLUE.get()), "#85C1D3");
    assertEquals(NorthemDark.hex(NorthemDark.BLUE.get()), "#87B2CE");
    assertEquals(NorthemDark.hex(NorthemDark.DARK_BLUE.get()), "#5B81AF");
    assertEquals(NorthemDark.hex(NorthemDark.RED.get()), "#E07473");
    assertEquals(NorthemDark.hex(NorthemDark.YELLOW.get()), "#FDCE84");
    assertEquals(NorthemDark.hex(NorthemDark.ORANGE.get()), "#F0A06F");
    assertEquals(NorthemDark.hex(NorthemDark.LIME.get()), "#C9CC78");
    assertEquals(NorthemDark.hex(NorthemDark.PURPLE.get()), "#C2A1CA");
  }

  @Test
  public void shouldReturnTheCorrectRGBColorCode() {
    assertEquals(NorthemDark.rgb(NorthemDark.DARKEST_COAL.get()), "rgb(31, 31, 35)");
    assertEquals(NorthemDark.rgb(NorthemDark.DARK_COAL.get()), "rgb(36, 36, 41)");
    assertEquals(NorthemDark.rgb(NorthemDark.COAL.get()), "rgb(41, 41, 46)");
    assertEquals(NorthemDark.rgb(NorthemDark.LIGHT_COAL.get()), "rgb(79, 79, 89)");
    assertEquals(NorthemDark.rgb(NorthemDark.DIRTY_WHITE.get()), "rgb(222, 222, 222)");
    assertEquals(NorthemDark.rgb(NorthemDark.WHITE.get()), "rgb(236, 238, 239)");
    assertEquals(NorthemDark.rgb(NorthemDark.AQUA.get()), "rgb(143, 206, 200)");
    assertEquals(NorthemDark.rgb(NorthemDark.LIGHT_BLUE.get()), "rgb(133, 193, 211)");
    assertEquals(NorthemDark.rgb(NorthemDark.BLUE.get()), "rgb(135, 178, 206)");
    assertEquals(NorthemDark.rgb(NorthemDark.DARK_BLUE.get()), "rgb(91, 129, 175)");
    assertEquals(NorthemDark.rgb(NorthemDark.RED.get()), "rgb(224, 116, 115)");
    assertEquals(NorthemDark.rgb(NorthemDark.YELLOW.get()), "rgb(253, 206, 132)");
    assertEquals(NorthemDark.rgb(NorthemDark.ORANGE.get()), "rgb(240, 160, 111)");
    assertEquals(NorthemDark.rgb(NorthemDark.LIME.get()), "rgb(201, 204, 120)");
    assertEquals(NorthemDark.rgb(NorthemDark.PURPLE.get()), "rgb(194, 161, 202)");
  }
}


src/main/java/com/arcticicestudio/northem/dark/package-info.java
/*
++++++++++++++++++++++++++++++++++++++++++++++
title     Package Info                       +
project   northem-dark-java                  +
file      package-info.java                  +
version   0.2.0                             +
author    Arctic Ice Studio                  +
email     development@arcticicestudio.com    +
website   http://arcticicestudio.com         +
copyright Copyright (C) 2016                 +
created   2016-01-23 15:44 UTC+0100          +
modified  2016-04-09 13:50 UTC+0200          +
++++++++++++++++++++++++++++++++++++++++++++++

[Description]
Root package of the "Northem Dark - Java" project, which implements the "Northem Dark" color palette as enumeration.
The package represents the public API.
The main entry point of the package is the "NorthemDark" class.

[Copyright]
Copyright (C) 2016 Arctic Ice Studio <development@arcticicestudio.com>

[References]
Northem Dark
  (https://github.com/arcticicestudio/northem-dark)
Java 8 API Documentation
  (https://docs.oracle.com/javase/8/docs/api/)
JavaFX 2.2 API Documentation
  (https://docs.oracle.com/javafx/2/api/)
JavaFX 2.2 Oracle Project Website
  (http://docs.oracle.com/javase/8/javase-clienttechnologies.htm)
Arctic Versioning Specification
  (http://specs.arcticicestudio.com/arcver) (ArcVer)
*/
/**
 * Root package of the <a href="https://github.com/arcticicestudio/northem-dark-java">Northem Dark - Java</a>
 * project, which implements the <a href="https://github.com/arcticicestudio/northem-dark">Northem Dark</a>
 * color palette as enumeration.
 * <p>
 *   The package represents the public API.
 *   The main entry point of the package is the {@link com.arcticicestudio.northem.dark.NorthemDark} class.
 * </p>
 *
 * @author Arctic Ice Studio &lt;development@arcticicestudio.com&gt;
 * @see <a href="https://github.com/arcticicestudio/northem-dark-java">Northem Dark - Java</a>
 * @see <a href="https://github.com/arcticicestudio/northem-dark">Northem Dark</a>
 * @version 0.2.0
 * @since 0.1.0
 */
package com.arcticicestudio.northem.dark;


src/main/java/com/arcticicestudio/northem/dark/NorthemDark.java
/*
+++++++++++++++++++++++++++++++++++++++++++++
title     Northem Dark Color Palette        +
project   northem-dark-java                 +
file      NorthemDark.java                  +
version   0.2.0                             +
author    Arctic Ice Studio                 +
email     development@arcticicestudio.com   +
website   http://arcticicestudio.com        +
copyright Copyright (C) 2016                +
created   2016-01-23 15:42 UTC+0100         +
modified  2016-04-09 14:05 UTC+0200         +
+++++++++++++++++++++++++++++++++++++++++++++

[Description]
"Northem Dark" color palette as enumeration for the "Northem Dark - Java" project.

[Copyright]
Copyright (C) 2016 Arctic Ice Studio <development@arcticicestudio.com>

[References]
Northem Dark
  (https://github.com/arcticicestudio/northem-dark)
Java 8 API Documentation
  (https://docs.oracle.com/javase/8/docs/api/)
JavaFX 2.2 API Documentation
  (https://docs.oracle.com/javafx/2/api/)
JavaFX 2.2 Oracle Project Website
  (http://docs.oracle.com/javase/8/javase-clienttechnologies.htm)
Arctic Versioning Specification
  (http://specs.arcticicestudio.com/arcver) (ArcVer)
*/
package com.arcticicestudio.northem.dark;

import javafx.scene.paint.Color;
import java.util.regex.Pattern;

/**
 * <a href="https://github.com/arcticicestudio/northem-dark">Northem Dark</a> color palette
  * as enumeration of the <a href="https://github.com/arcticicestudio/northem-dark-java">Northem Dark - Java</a> project.
 * <div>
 *   <table style="display:inline-block">
 *     <thead>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(31,31,35);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(36,36,41);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(41,41,46);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(79,79,89);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(222,222,222);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(236,238,239);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(143,206,200);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(133,193,211);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(135,178,206);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(91,129,175);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(224,116,115);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(253,206,132);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(240,160,111);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(201,204,120);margin: 0"></th>
 *       <th style="border:none;width:25px;height:30px;background-color:rgb(194,161,202);margin: 0"></th>
 *     </thead>
 *   </table>
 * </div>
 *
 * @author Arctic Ice Studio &lt;development@arcticicestudio.com&gt;
 * @see <a href="https://github.com/arcticicestudio/northem-dark">"Northem Dark" GitHub Repository</a>
 * @see <a href="https://github.com/arcticicestudio/northem-dark-java">"Northem Dark - Java" GitHub Repository</a>
 * @since 0.1.0
 */
public enum NorthemDark {
  /**
   * The color {@code darkest-coal} with an RGB value of {@code rgb(31,31,35)} and HEX value of {@code #1F1F23}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(31,31,35);margin: 0"></div>
   */
  DARKEST_COAL(31,31,35),
  /**
   * The color {@code dark-coal} with an RGB value of {@code rgb(36,36,41)} and HEX value of {@code #242429}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(36,36,41);margin: 0"></div>
   */
  DARK_COAL(36,36,41),
  /**
   * The color {@code coal} with an RGB value of {@code rgb(41,41,46)} and HEX value of {@code #29292E}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(41,41,46);margin: 0"></div>
   */
  COAL(41,41,46),
  /**
   * The color {@code light-coal} with an RGB value of {@code rgb(79,79,89)} and HEX value of {@code #4F4F59}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(79,79,89);margin: 0"></div>
   */
  LIGHT_COAL(79,79,89),
  /**
   * The color {@code dirty-white} with an RGB value of {@code rgb(222,222,222)} and HEX value of {@code #DEDEDE}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(222,222,222);margin: 0"></div>
   */
  DIRTY_WHITE(222,222,222),
  /**
   * The color {@code white} with an RGB value of {@code rgb(236,238,239)} and HEX value of {@code #ECEEEF}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(236,238,239);margin: 0"></div>
   */
  WHITE(236,238,239),
  /**
   * The color {@code aqua} with an RGB value of {@code rgb(143,206,200)} and HEX value of {@code #8FCEC8}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(143,206,200);margin: 0"></div>
   */
  AQUA(143,206,200),
  /**
   * The color {@code light-blue} with an RGB value of {@code rgb(133,193,211)} and HEX value of {@code #85C1D3}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(133,193,211);margin: 0"></div>
   */
  LIGHT_BLUE(133,193,211),
  /**
   * The color {@code blue} with an RGB value of {@code rgb(135,178,206)} and HEX value of {@code #87B2CE}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(135,178,206);margin: 0"></div>
   */
  BLUE(135,178,206),
  /**
   * The color {@code dark-blue} with an RGB value of {@code rgb(91,129,175)} and HEX value of {@code #5B81AF}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(91,129,175);margin: 0"></div>
   */
  DARK_BLUE(91,129,175),
  /**
   * The color {@code red} with an RGB value of {@code rgb(224,116,115)} and HEX value of {@code #E07473}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(224,116,115);margin: 0"></div>
   */
  RED(224,116,115),
  /**
   * The color {@code yellow} with an RGB value of {@code rgb(253,206,132)} and HEX value of {@code #FDCE84}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(253,206,132);margin: 0"></div>
   */
  YELLOW(253,206,132),
  /**
   * The color {@code orange} with an RGB value of {@code rgb(240,160,111)} and HEX value of {@code #F0A06F}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(240,160,111);margin: 0"></div>
   */
  ORANGE(240,160,111),
  /**
   * The color {@code lime} with an RGB value of {@code rgb(201,204,120)} and HEX value of {@code #C9CC78}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(201,204,120);margin: 0"></div>
   */
  LIME(201,204,120),
  /**
   * The color {@code purple} with an RGB value of {@code rgb(194,161,202)} and HEX value of {@code #C2A1CA}.
   * <div style="border:none;width:25px;height:30px;background-color:rgb(194,161,202);margin: 0"></div>
   */
  PURPLE(194,161,202);

  /**
   * Compiled pattern to convert the RGB and HEX color values.
   */
  private static final Pattern COLOR_PREFIX = Pattern.compile("0x", Pattern.LITERAL);

  /**
   * Contains the RGB color code.
   */
  private final Color COLOR;

  /**
   * Constructs a new color object.
   *
   * @param RED The value for the additive primary color {@code red}
   * @param GREEN The value for the additive primary color {@code green}
   * @param BLUE The value for the additive primary color {@code blue}
   */
  NorthemDark(final int RED, final int GREEN, final int BLUE) {
    COLOR = Color.rgb(RED, GREEN, BLUE);
  }

  /**
   * Converts the color to the {@code RGB} identifier.
   *
   * @param COLOR The color object
   * @return a string representation of the {@code RGB} identifier
   */
  public static String rgb(final Color COLOR) {
    String hex = COLOR_PREFIX.matcher(COLOR.toString()).replaceAll("");
    String hexRed = hex.substring(0, 2).toUpperCase();
    String hexGreen = hex.substring(2, 4).toUpperCase();
    String hexBlue = hex.substring(4, 6).toUpperCase();

    String intRed = Integer.toString(Integer.parseInt(hexRed, 16));
    String intGreen = Integer.toString(Integer.parseInt(hexGreen, 16));
    String intBlue = Integer.toString(Integer.parseInt(hexBlue, 16));

    return String.join("", "rgb(", intRed, ", ", intGreen, ", ", intBlue, ")");
  }

  /**
   * Converts the color to the {@code HEX} identifier.
   *
   * @param COLOR The color object
   * @return a string representation of the {@code HEX} identifier
   */
  public static String hex(final Color COLOR) {
    return COLOR_PREFIX.matcher(COLOR.toString()).replaceAll("#").substring(0, 7).toUpperCase();
  }

  /**
   * Returns the color object.
   *
   * @return the color object
   */
  public Color get() {
    return COLOR;
  }
}
